cmake/compiler/gcc/target.cmake

```diff
@@ -6,10 +6,36 @@ set_ifndef(C++ g++)
 # GCC-based toolchains
 
 if("${SPARSE}" STREQUAL "y")
-  find_program(CMAKE_C_COMPILER cgcc)
-  find_program(REAL_CC ${CROSS_COMPILE}${CC} PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
-  set(ENV{REAL_CC} ${REAL_CC})
-else()
+  # No search PATHS because we need more than cgcc in the default PATH
+  find_program(CMAKE_C_COMPILER cgcc REQUIRED)
+  message(STATUS "Found sparse: ${CMAKE_C_COMPILER}")
+
+  find_program(SPARSE_REAL_COMPILER ${CROSS_COMPILE}${CC} PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH REQUIRED)
+
+  # We know what REAL_CC _must_ be, so why do we ask the user to define it?  Because CMake
+  # cannot set (evil) build time env variables at configure time:
+  # https://gitlab.kitware.com/cmake/community/-/wikis/FAQ#how-can-i-get-or-set-environment-variables
+  # As of Sep. 2022, sparse/cgcc has unfortunately no --real-cc option as it should.
+  #
+  # In theory we could define REAL_CC ourselves here and fix the configure-time
+  # --print-file-name below (and maybe others). Then ask the user to define REAL_CC later
+  # at _build_ time. But in practice who would define environment variables at build time
+  # _only_?  So best to fail early and clearly here.
```

Comment:
Yes, unfortunately there is no way to define new environmental variable to parent process, CMake does it only for child processes. It would be possible if we would have invocation of sparse inside CMake somewhere but as far as I understand we would like to launch it manually yes?

Reply:
Thanks for the review!

> It would be possible if we would have invocation of sparse inside CMake somewhere 

`sparse/cgcc` pretends to be the `CMAKE_C_COMPILER` which is used both at cmake time and ninja time but mainly at ninja time. As explained at https://gitlab.kitware.com/cmake/community/-/wikis/FAQ#how-can-i-get-or-set-environment-variables and https://github.com/zephyrproject-rtos/zephyr/pull/43776#pullrequestreview-1023060702 there's no way to "pass" environment variables to `CMAKE_C_COMPILER` through ninja.

I think a new `sparse/cgcc --real-cc` flag would be the start of the best fix but we can't wait that long, sparse results are too critical. I don't know how much zephyr/cmake refactoring `--real-cc` would/will require.